src/docker/composeappengine.cc

```diff
@@ -22,22 +24,13 @@ AppEngine::Result ComposeAppEngine::fetch(const App& app) {
   try {
     AppState state(app, appRoot(app), true);
 
-    if (!download(app)) {
-      state.setState(AppState::State::kDownloadFailed);
-      return result;
-    }
+    download(app);
```

Comment:
this is hard for me to reason about. we used to set the state fail to kDownloadFailed and now we aren't. I assume this means setting that file to failed states never meant anything useful? i.e we only look at it for good states?

Reply:
Yes, those *Failed states are useless, I'll remove them.

Reply:
Removed the failed states.